### PR TITLE
Attempt to port to rustls 0.20

### PR DESCRIPTION
**Pull request by mfelsche**
This is the continuation of #47 dressed up as a new PR. This is based on the heavy lifting done by @silverjam and some minor contributions by me.

This updates the certs used for testing as some were expired. I added a script based on openssl to regenerate the ones i updated. They are valid for 10 years.

This obsoletes #47

### CODE REVIEW

**Comment by NathanRoyer** [comment]
Thanks for your work; this allowed me to make `mail-send` work with `async-std`.

Is there anything blocking the merge?

**Comment by mfelsche** [comment]
I honestly don't know.

**Comment by silverjam** [comment]
> Thanks for your work; this allowed me to make `mail-send` work with `async-std`.
> 
> 
> 
> Is there anything blocking the merge?

The main readme of this crate states that @skade is looking for a new maintainer (https://github.com/async-rs/async-tls/blob/master/README.md), so that's probably all that's blocking things.

**Comment by silverjam** [comment]
Maybe @dignifiedquire or @yoshuawuyts could help get this merged?

**Comment by Fishrock123** [comment]
@mfelsche Have you tried contacting Florian via the email in the maintenance notice? https://github.com/async-rs/async-tls#maintenance-notice

**Comment by dignifiedquire** [comment]
I will try to review latest next week

File: examples/server/src/main.rs
Comment (dignifiedquire) [inline examples/server/src/main.rs:6]: why do you need this over https://docs.rs/async-std/latest/async_std/io/trait.WriteExt.html ?

**Review by dignifiedquire — APPROVED** [review]
looks pretty good to me

File: examples/server/src/main.rs
Comment (mfelsche) [reply examples/server/src/main.rs:6]: `AsyncWriteExt` from the `futures_util` crate gives us `.close()` on the `TlsStream` which implements `AsyncWrite`. See: https://docs.rs/futures-util/latest/futures_util/io/trait.AsyncWriteExt.html#method.close

Sadly `WriteExt` from async_std doesn't provide `.close()`. See this related issue: https://github.com/async-rs/async-std/issues/977

Comment (mfelsche) [reply examples/server/src/main.rs:6]: Sorry for getting back to you so late. Your review slipped through my notifications.

Comment (dignifiedquire) [reply examples/server/src/main.rs:6]: I see, thanks for the info

**Comment by dignifiedquire** [comment]
Thanks everyone, will cut a release tomorrow!

**Comment by silverjam** [comment]
Thank you @dignifiedquire!

**Comment by dignifiedquire** [comment]
Release `0.12.0` is out

**Pull request merged**
